pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/agent/impl/DockerPipelineFromDockerfile.java

```diff
@@ -52,6 +55,33 @@ public void setFilename(String filename) {
         this.filename = filename;
     }
 
+    public @Nonnull String getDir() {
```

Comment:
This is a lie; `dir` will always be `null` until `setDir` has been called, and even then it allows `null` values to be set.

I guess you intended to put `@Nonnull` on `getActualDir`?